### PR TITLE
support config leader election

### PR DESCRIPTION
**Pull request by Frapschen**
The same as https://github.com/operator-framework/operator-sdk/issues/1813#issuecomment-733513350, I encounter `leader election lost`. 

We can increasing the `LeaseDuration`, `RenewDeadline ` and `RetryPeriod` like [opentelemetry-operator](https://github.com/open-telemetry/opentelemetry-operator/blob/a8653601cd6a6e2b35fd7f3e1a28b4e9608fb794/main.go#L181)

### CODE REVIEW

**Comment by JaredTan95** [comment]
you can also upgrade the helm chart.

**Comment by sumo-drosiek** [comment]
@Frapschen thank you for the PR

Could you move configuration to `config.go`? We would like to use configuration file instead of command line arguments. Also I don't see a reason to expose all command line arguments. We can limit to the election ones

> you can also upgrade the helm chart.

We are releasing as separate pull requests:

https://github.com/SumoLogic/tailing-sidecar/blob/a1fd0a8f32f5ff7ce790d5b717175152999cf95b/docs/release.md

**Comment by JaredTan95** [comment]
> We are releasing as separate pull requests:
> https://github.com/SumoLogic/tailing-sidecar/blob/a1fd0a8f32f5ff7ce790d5b717175152999cf95b/docs/release.md

OK, I mean that should also upgrade the content `templates/*.yaml` in the chart, not bump up chart version:-P

**Review by JaredTan95 — APPROVED** [review]
LGTM

File: helm/tailing-sidecar-operator/conf/operator.yaml [hunk 1]
```diff
@@ -1,2 +1,6 @@
 sidecar:
   image: {{ .Values.sidecar.image.repository }}:{{ .Values.sidecar.image.tag | default .Chart.AppVersion }}
+leaderElection:
+  leaseDuration: {{ .Values.operator.leaderElection.leaseDuration }}
+  renewDeadline: {{ .Values.operator.leaderElection.renewDeadline }}
+  retryPeriod: {{ .Values.operator.leaderElection.retryPeriod }}
```
Comment (sumo-drosiek) [inline helm/tailing-sidecar-operator/conf/operator.yaml:6]:
```suggestion
  retryPeriod: {{ .Values.operator.leaderElection.retryPeriod }}

```

File: operator/config.go [hunk 2]
```diff
@@ -2,24 +2,37 @@ package main
...
 	// Set default values
 	config := Config{
 		Sidecar: SidecarConfig{
 			Image: "sumologic/tailing-sidecar:latest",
 		},
+		leaderElection: leaderElectionConfig{
+			LeaseDuration: time.Second * 137,
+			RenewDeadline: time.Second * 107,
+			RetryPeriod:   time.Second * 26,
+		},
```
Comment (sumo-drosiek) [inline operator/config.go:35]:
```suggestion
		// reference for values: https://github.com/open-telemetry/opentelemetry-operator/blob/a8653601cd6a6e2b35fd7f3e1a28b4e9608fb794/main.go#L181
		leaderElection: leaderElectionConfig{
			LeaseDuration: time.Second * 137,
			RenewDeadline: time.Second * 107,
			RetryPeriod:   time.Second * 26,
		},
```

File: operator/config_test.go [hunk 3]
```diff
@@ -36,18 +42,31 @@ sidecar:
...
 		},
 		{
 			name: "overwrite defaults",
 			content: `
 sidecar:
-  image: my-new-image`,
+  image: my-new-image
+leaseDuration: 10s
+renewDeadline: 10s
+retryPeriod: 10s`,
```
Comment (sumo-drosiek) [inline operator/config_test.go:60]:
```suggestion
leaderElection:
  leaseDuration: 10s
  renewDeadline: 10s
  retryPeriod: 10s`,
```

Comment (sumo-drosiek) [reply operator/config_test.go:60]: tests are failing due to that

**Review by sumo-drosiek — APPROVED** [review]

**Pull request merged**